examples/cluster/plot_kmeans_silhouette_analysis.py

```diff
@@ -0,0 +1,139 @@
+"""
+===============================================================================
+Silhouette analysis for sample data clustered using KMeans clustering algorithm
+===============================================================================
+
+Silhouette analysis can be used to study the separation distance between the
+resulting clusters. The silhouette plot displays a measure of how close each
+point in one cluster is to points in the neighboring clusters. This measure has
+a range of [-1, 1]. Silhoette coefficients (as these values are referred to as)
+near +1 indicate that the sample is far away from the neighboring clusters.
+A value of 0 indicates that the sample is on or very close to the decision 
+boundary between two neighboring clusters and negative values (upto -1) 
+indicate that those samples might have been assigned to the wrong cluster.
+"""
```

Comment:
I have wondered if this introduction should give more direct insight into the chosen example, e.g. to state that the silhouette plot shows that 6 clusters is clearly a bad pick, but is more ambivalent in deciding between 2 and 4 (I also wonder whether showing 3 and 5 would be still more instructive, or merely overkill).